client/components/social-buttons/facebook.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import SocialLogo from 'social-logos';
+
+export default class FacebookLoginButton extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			FB: null,
+		};
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentDidMount() {
+		this.loadDependency().then( FB => {
+			FB.init( {
+				appId: this.props.appId,
+				xfbml: this.props.xfbml,
+				cookie: this.props.cookie,
+				version: this.props.version,
+			} );
+			this.setState( { FB } );
+		} );
+	}
+
+	loadDependency() {
+		return new Promise( resolve => {
+			if ( window.FB ) {
+				resolve( window.FB );
+			} else {
+				( function ( d, s, id ) {
+					const element = d.getElementsByTagName( s )[0];
+					const fjs = element;
+					let js = element;
+					if ( d.getElementById( id ) ) { return; }
+					js = d.createElement( s ); js.id = id;
+					js.src = '//connect.facebook.net/en_US/sdk.js';
+					fjs.parentNode.insertBefore( js, fjs );
```

Comment:
In the past, we've used [`loadScript`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/load-script/index.js#L13) for loading third party scripts like this, as it is a bit clearer than the script Facebook asks us to copy/paste into the app. What do you think about using it here?